dataactbroker/handlers/fileHandler.py

```diff
@@ -494,31 +498,112 @@ def startGenerationJob(self, submission_id, file_type):
 
         job = jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, file_type_name, "file_upload")
         job.filename = upload_file_name
+        job.original_filename = timestamped_name
         job.job_status_id = jobDb.getJobStatusId("running")
         jobDb.session.commit()
         if file_type in ["D1", "D2"]:
             try:
                 valJob = jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, file_type_name, "csv_record_validation")
                 valJob.filename = upload_file_name
+                valJob.original_filename = timestamped_name
                 valJob.job_status_id = jobDb.getJobStatusId("waiting")
                 job.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
                 job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
                 valJob.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
                 valJob.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
+                # Generate random uuid and store generation task
+                task_key = uuid4()
+                task = FileGenerationTask(generation_task_key = task_key, submission_id = submission_id, file_type_id = jobDb.getFileTypeId(file_type_name))
+                jobDb.session.add(task)
+
                 jobDb.session.commit()
             except ValueError as e:
                 # Date was not in expected format
                 exc = ResponseException(str(e),StatusCode.CLIENT_ERROR,ValueError)
                 return False, JsonResponse.error(exc, exc.status, url = "", start = "", end = "",  file_type = file_type)
-            get_url = CONFIG_BROKER["".join([file_type_name, "_url"])].format(cgac_code, start_date, end_date)
-            jq.generate_d_file.delay(get_url, user_id, job.job_id, InterfaceHolder, timestamped_name, self.isLocal)
+            # Create file D API URL with dates and callback URL
+            callback = "http://{}:{}/v1/complete_generation/{}/".format(CONFIG_SERVICES["broker_api_host"], CONFIG_SERVICES["broker_api_port"],task_key)
+            get_url = CONFIG_BROKER["".join([file_type_name, "_url"])].format(cgac_code, start_date, end_date, callback)
+            if not self.call_d_file_api(get_url):
+                # No results found, mark job invalid
+                jobDb.markJobStatus(job.job_id,"invalid")
+                job.error_message = "No results found for that date range"
+                jobDb.session.commit()
         else:
             # TODO add generate calls for E and F
             jobDb.markJobStatus(job.job_id,"finished")
+
             pass
 
         return True, None
 
+    def get_xml_response_content(self, api_url):
+        """ Retrieve XML Response from the provided API url """
+        return requests.get(api_url, verify=False).text
```

Comment:
This part has always come back so far, but that's not a bad idea to add.  I'll set to 20 seconds for now.